File: app/src/main/java/com/rkbapps/neetflix/fragments/PersonImageFragment.java
package com.rkbapps.neetflix.fragments;

import android.annotation.SuppressLint;
import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.rkbapps.neetflix.R;
import com.rkbapps.neetflix.adapter.PersonImageAdapter;
import com.rkbapps.neetflix.models.images.PersonImageModel;
import com.rkbapps.neetflix.services.ApiData;
import com.rkbapps.neetflix.services.PersonApi;
import com.rkbapps.neetflix.services.RetrofitInstance;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;


public class PersonImageFragment extends Fragment {

    public PersonImageFragment() {
        // Required empty public constructor
    }
    private RecyclerView recyclerView;
    @SuppressLint("MissingInflatedId")
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_person_image, container, false);
        int id = getArguments().getInt("id");

        recyclerView = view.findViewById(R.id.recyclerPersonImage);
        recyclerView.setLayoutManager(new GridLayoutManager(getContext(),2));

        PersonApi personApi = RetrofitInstance.getPersonApi();
        Call<PersonImageModel>responseCall = personApi.getPersonImages(id, ApiData.API_KEY);
        responseCall.enqueue(new Callback<PersonImageModel>() {
            @Override
            public void onResponse(Call<PersonImageModel> call, Response<PersonImageModel> response) {
                if(response.isSuccessful())
                    recyclerView.setAdapter(new PersonImageAdapter(getContext(),response.body().getProfiles()));
            }

            @Override
            public void onFailure(Call<PersonImageModel> call, Throwable t) {

            }
        });


        return view;
    }
}

File: app/src/main/java/com/rkbapps/neetflix/fragments/person/PersonImageFragment.kt
package com.rkbapps.neetflix.fragments.person

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.GridLayoutManager
import com.rkbapps.neetflix.R
import com.rkbapps.neetflix.adapter.person.PersonImageAdapter
import com.rkbapps.neetflix.databinding.FragmentPersonImageBinding
import com.rkbapps.neetflix.repository.person.PersonImageRepository
import com.rkbapps.neetflix.utils.Resource
import com.rkbapps.neetflix.viewmodelfactories.person.PersonImageViewModelFactory
import com.rkbapps.neetflix.viewmodels.person.PersonImageViewModel

class PersonImageFragment : Fragment() {

    private lateinit var binding: FragmentPersonImageBinding
    private lateinit var viewModel: PersonImageViewModel


    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        binding =
            DataBindingUtil.inflate(inflater, R.layout.fragment_person_image, container, false)
        val id = requireArguments().getInt("id")
        val repository = PersonImageRepository()
        viewModel = ViewModelProvider(
            requireActivity(),
            PersonImageViewModelFactory(repository, id)
        )[PersonImageViewModel::class.java]

        binding.recyclerPersonImage.layoutManager = GridLayoutManager(context, 2)

        loadImages()
        return binding.root
    }

    private fun loadImages() {

        viewModel.images.observe(viewLifecycleOwner, Observer {
            when (it) {

                is Resource.Loading -> {

                }

                is Resource.Success -> {
                    if (it.data?.body() != null)
                        binding.recyclerPersonImage.adapter =
                            PersonImageAdapter(requireContext(), it.data.body()!!.profiles);
                }

                is Resource.Error -> {

                }
            }
        })


    }


}